tests/commands/run/test_integration.py

```diff
@@ -1,21 +1,135 @@
-import tempfile, shutil
+import yaml, pytest
 from ...util import *
+from ...fixtures import *
 from .util import *
 from sinol_make import configure_parsers
 
-def test_run_simple():
-	with tempfile.TemporaryDirectory() as tmpdir:
-		package_path = os.path.join(tmpdir, "abc")
-		shutil.copytree(get_simple_package_path(), package_path)
+def test_simple(create_package):
+	"""
+	Test a simple run.
+	"""
+	package_path = create_package
+	command = get_command()
+	create_ins_outs(package_path, command)
 
-		command = get_command()
-		create_ins(package_path, command)
-		os.chdir(package_path)
-		create_outs(package_path, command)
-		os.chdir(package_path)
+	parser = configure_parsers()
 
-		parser = configure_parsers()
+	args = parser.parse_args(["run"])
+	command = Command()
+	command.run(args)
 
-		args = parser.parse_args(["run"])
-		command = Command()
+
+def test_no_expected_scores(capsys, create_package):
+	"""
+	Test with no sinol_expected_scores in config.yml.
+	Should run, but exit with exit code 1.
+	"""
+	package_path = create_package
+	command = get_command()
+	create_ins_outs(package_path, command)
+
+	config_path = os.path.join(package_path, "config.yml")
+	config = yaml.load(open(config_path, "r"), Loader=yaml.SafeLoader)
+	del config["sinol_expected_scores"]
+	open(config_path, "w").write(yaml.dump(config))
+
+	parser = configure_parsers()
+	args = parser.parse_args(["run"])
+	command = Command()
+	with pytest.raises(SystemExit) as e:
 		command.run(args)
+
+	assert e.type == SystemExit
+	assert e.value.code == 1
+
+	out = capsys.readouterr().out
+	assert "Solutions were added:" in out
+	assert "abc.cpp" in out
+
+
+def test_apply_suggestions(create_package):
+	"""
+	Test with no sinol_expected_scores in config.yml.
+	Verifies that suggestions are applied.
```

Comment:
We can extend this comment, currently it doesn't say much more than the function name.

Reply:
I changed some comments: 32873ce9a2f8f9c627efa0c1dc127260b0c08569